Card/src/main/java/com/paypal/android/card/CardClient.kt

```diff
@@ -1,42 +1,155 @@
+    private var confirmPaymentSourceResponse: ConfirmPaymentSourceResponse? = null
+    private var lifeCycleObserver: CardLifeCycleObserver? = null
 
     /**
-     * Confirm [Card] payment source for an order. Use this method for Kotlin integrations
+     *  CardClient constructor
      *
-     * @param request [CardRequest] for requesting an order approval
+     *  @param [activity] Activity that launches the card client
+     *  @param [configuration] Configuration parameters for client
+     *  @param [coroutineContext] CoroutineContext to execute
      */
-    suspend fun approveOrder(request: CardRequest): CardResult =
-        cardAPI.confirmPaymentSource(request.orderID, request.card)
+    @JvmOverloads
+    constructor(
+        activity: FragmentActivity,
+        configuration: CoreConfig,
+        coroutineContext: CoroutineContext = Dispatchers.Main
+    ) : this(activity, CardAPI(API(configuration)), coroutineContext)
 
     /**
-     * Confirm [Card] payment source for an order. Use this method for Java integrations
+     * Confirm [Card] payment source for an order.
      *
-     * @param request [CardRequest] for requesting an order approval
-     * @param callback callback to get response
+     * @param orderId [String] order id to confirm
+     * @param cardRequest [CardRequest] for requesting an order approval
+     * @param threeDSecureRequest [threeDSecureRequest] to run transaction with 3DS
+     * @param callback [ApproveOrderCallback] callback for responses
      */
-    fun approveOrder(request: CardRequest, callback: ApproveOrderCallback) {
-        CoroutineScope(Dispatchers.IO).launch {
-            try {
-                val result = approveOrder(request)
+    @JvmOverloads
+    fun approveOrder(
+        orderId: String,
+        cardRequest: CardRequest,
+        threeDSecureRequest: ThreeDSecureRequest? = null,
+        callback: ApproveOrderCallback
+    ) {
+        CoroutineScope(coroutineContext).launch {
+            confirmPaymentSource(orderId, cardRequest, threeDSecureRequest, callback)
+        }
+    }
+
+    /**
+     * Creates order and confirms [Card] payment source for said order. Its private until phase 2
+     *
+     * @param orderRequest [String] order to create
+     * @param cardRequest [CardRequest] for requesting an order approval
+     * @param threeDSecureRequest [threeDSecureRequest] to run transaction with 3DS
+     * @param callback [ApproveOrderCallback] callback for responses
+     */
+    private fun createAndApproveOrder(
+        orderRequest: OrderRequest,
+        cardRequest: CardRequest,
+        threeDSecureRequest: ThreeDSecureRequest? = null,
+        callback: ApproveOrderCallback
+    ) {
+        CoroutineScope(coroutineContext).launch {
+            val orderId = createOrder(orderRequest, threeDSecureRequest)
+            confirmPaymentSource(orderId, cardRequest, threeDSecureRequest, callback)
+        }
+    }
+
+    private suspend fun createOrder(
```

Comment:
I think this can be removed as well. It's only used by the method above.

Reply:
same awnser es `createAndApproveOrder` being private